### PR TITLE
Make Sortition Hash compliant with specification

### LINKED ISSUE #907: Make Sortition Hash compliant with specification
**Issue**
`create_sortition_hash()` returns $H(round||counter||step||seed)$, but the [specifications](https://wiki.dusk.network/en/tech/consensus/deterministic-sortition) state it should be $H(seed||round||step||counter)$. 

**Solution**
Change hashing order to the one in specifications

### PR DESCRIPTION
**Pull request by fed-franz**
Resolves #907 

This PR changes the sortition hash from $H(round||counter||step||seed)$ to $H(seed||round||step||counter)$, 
to be compliant with Deterministic Sortition specification.

The following tests are also updated: `test_sortition_hash`, `test_generate_sortition_score`, `test_collect_votes`, `test_deterministic_sortition_1`, `test_deterministic_sortition_2`.

Note that `test_collect_votes` code has been also overhauled to make it simpler and easier to update in the future.

| :exclamation:  IMPORTANT: This is a BREAKING CHANGE |
|-|

### CODE REVIEW

**Comment by fed-franz** [comment]
IMHO, `test_collect_votes` should use a committee size of 64 instead of 10

**Review by herr-seppia — DISMISSED** [review]
Since we are aiming to test compatibility between node and dusk-blockchain I'm gonna to block this merge until the same changes are reflected on dusk-blockchain too

**Comment by herr-seppia** [comment]
@fed-franz I've dismissed my blocking review. Please go on if this is still required

**Comment by fed-franz** [comment]
Does this mean the mirror issue on `dusk-blockchain` is not relevant anymore?

- https://github.com/dusk-network/dusk-blockchain/issues/1494

**Review by herr-seppia — APPROVED** [review]
LGTM

**Pull request merged**
